pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -885,6 +887,43 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         applyNCT(arr, num_qubits, {}, {}, wires, inverse);
     }
 
+    template <class PrecisionT>
+    static void
+    applyNCSX(std::complex<PrecisionT> *arr, const size_t num_qubits,
+              const std::vector<size_t> &controlled_wires,
+              const std::vector<bool> &controlled_values,
+              const std::vector<size_t> &wires, const bool inverse) {
+        using ParamT = PrecisionT;
+
+        const PrecisionT half = 0.5;
+        const std::complex<PrecisionT> z0{half, (inverse) ? -half : half};
+        const std::complex<PrecisionT> z1 = std::conj(z0);
```

Comment:
```suggestion
        constexpr PrecisionT half = 0.5;
        constexpr std::complex<PrecisionT> z0{half, (inverse) ? -half : half};
        constexpr std::complex<PrecisionT> z1 = std::conj(z0);
```

Reply:
Even in situations where the compiler will probably not be able to replace with a literal in compile time it is OK to set as `constexpr`. You can only make the obvious one a constexpr if you prefer.

Reply:
Makes sense! Thanks!